lib/galaxy/web/framework/helpers/grids.py

```diff
@@ -160,13 +162,13 @@ def loads_recurse(item):
                     if isinstance( column_filter, list ):
                         # Filter is a list; process each item.
                         for filter in column_filter:
-                            if not isinstance( filter, basestring ):
-                                filter = unicode( filter ).encode("utf-8")
+                            if not isinstance( filter, string_types ):
+                                text_type( filter ).encode("utf-8")
```

Comment:
list comprehension was my first thought too, but in your version we're dropping `filter` if it is of `string_type`.
how about this?
`column_filter = [text_type(_).encode('utf-8') if not isinstance(_, string_types) else _ for _ in column_filter)]`